tripy/tests/spec_verification/test_dtype_constraints.py

```diff
@@ -15,120 +15,168 @@
+                        )
+                    )
+                else:
+                    neg_func_list.append(
+                        (
+                            func_name,
+                            func_obj,
+                            inputs,
+                            return_dtype,
+                            namespace,
+                            func_name + "_invalid: " + ", ".join(ids),
+                        )
+                    )
+                if exception:
+                    positive_case = True
 
 
-@pytest.mark.parametrize("test_data", func_list, ids=lambda val: val[5])
-def test_dtype_constraints(test_data):
-    func_obj, inputs, returns, namespace, positive_case, _ = test_data
-    # create all inputs for positive test case
+def _run_dtype_constraints_subtest(test_data):
+    func_name, func_obj, inputs, _, namespace, _ = test_data
     kwargs = {}
-    for param_name, input_desc in inputs.items():
-        kwargs[param_name] = create_obj(func_obj, param_name, input_desc, namespace)
-    # run api call
+    # Create all input objects using object_builders.create_obj.
+    for param_name, param_type in inputs.items():
+        kwargs[param_name] = create_obj(func_obj, func_name, param_name, param_type, namespace)
+    # Run api call through _method_handler and setup namespace (api_call_locals).
     api_call_locals = {"kwargs": kwargs}
-    exec(method_handler(kwargs, func_obj), globals(), api_call_locals)
-    # assert with output
-    if positive_case:
-        assert api_call_locals[RETURN_VALUE].dtype == namespace[list(returns.values())[0]["dtype"]]
-    else:
-        assert (
-            not api_call_locals[RETURN_VALUE].dtype == namespace[list(returns.values())[0]["dtype"]]
-        ), f"{fg('red')}invalid test case failure: {list(namespace.values())[0]} is a valid dtype for {list(namespace.keys())[0]}{attr('reset')}"
+    _method_handler(func_name, kwargs, func_obj, api_call_locals)
+    # If output does not have dtype skip .eval().
+    if isinstance(api_call_locals[RETURN_VALUE], int):
+        return api_call_locals, namespace
+    # If output is a list then checking the return the first element in the list. (Assumes list of Tensors)
+    if isinstance(api_call_locals[RETURN_VALUE], List):
+        api_call_locals[RETURN_VALUE] = api_call_locals[RETURN_VALUE][0]
+    # Run eval to check for any backend errors.
+    api_call_locals[RETURN_VALUE].eval()
+    return api_call_locals, namespace
+
+
+# # Only run positive test cases for L0 testing.
+# @pytest.mark.parametrize("test_data", pos_func_list, ids=lambda val: val[5])
+# def test_pos_dtype_constraints(test_data):
+#     _, _, _, return_dtype, _, _ = test_data
+#     api_call_locals, namespace = _run_dtype_constraints_subtest(test_data)
+#     if isinstance(api_call_locals[RETURN_VALUE], tp.Tensor):
+#         assert api_call_locals[RETURN_VALUE].dtype == namespace[return_dtype]
+
```

Comment:
Are these tests intentionally commented out?

Reply:
I was testing L0 timings during the night with different configurations

Reply:
I see. It might be better to stash experimental changes locally rather than committing them to the branch. Otherwise it's possible they'll be accidentally merged.

Reply:
Should I just make another branch then if I want to test CI machine timing?

Reply:
Yeah, that's probably cleaner. Although if we just move these tests to L1 then it's probably not necessary.